src/utils/handleResponseRandomization.ts

```diff
@@ -1,5 +1,18 @@
 import { IndividualComponent, StringOption } from '../parser/types';
 
+export function randomizeForm(componentConfig: IndividualComponent) {
+  return componentConfig.response.reduce((acc, response) => {
+    if (componentConfig.randomizeForm) {
+      const shuffled = [...componentConfig.response]
+        .map((value) => ({ value, sort: Math.random() }))
+        .sort((a, b) => a.sort - b.sort)
+        .map(({ value }) => value);
+      acc[response.id] = [shuffled.findIndex((value) => value.id === response.id)];
+    }
+    return acc;
+  }, {} as Record<string, number[]>);
+}
```

Comment:
This looks buggy. For each item in the response (each question) we produce a new shuffle and then find the index in the new shuffle. I don't think that's quite what we want.

I think you only need this
```
const shuffled = [...componentConfig.response]
        .map((value) => ({ value, sort: Math.random() }))
        .sort((a, b) => a.sort - b.sort)
        .map(({ value }) => value);
```